### ISSUE TITLE
Cannot set the CFBundleDevelopmentRegion in Info.plist of resource bundles

### ISSUE THREAD

**Issue by xingheng** [CocoaPods/CocoaPods#7486]
* [*] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Clone the lastest CocoaPods project to review the resource bundle's Info.plist's generate rules.
Check the issue list.

## What did you expect to happen?

Apply the app target's `development region` to all the pods' resource bundle.

## What happened instead?

The `CFBundleDevelopmentRegion` key in Info.plist is set to `en` with a default key.

### Stack

```
   CocoaPods : 1.3.1
        Ruby : ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin16]
    RubyGems : 2.6.13
        Host : Mac OS X 10.12.6 (16G29)
       Xcode : 9.2 (9C40b)
         Git : git version 2.14.3 (Apple Git-98)
Ruby lib dir : /Users/hanwei/.rvm/rubies/ruby-2.4.1/lib
Repositories : bitbucket - https://bitbucket.org/UItachi/specs.git @ 6aa9c96ed1c82b32f906285dc004c0a50bc16666
               master - https://github.com/CocoaPods/Specs.git @ bec882057e88704a796126484d0e627b654b43eb
```

### Installation Source

```
Executable Path: /Users/hanwei/.rvm/gems/ruby-2.4.1/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.0
cocoapods-try         : 1.1.0
```
...

## Project that demonstrates the issue

I found the [onepassword-app-extension](https://github.com/agilebits/onepassword-app-extension) has the same issue there.

---

I found [this issue](https://github.com/CocoaPods/Xcodeproj/issues/259) is the starter, resource bundle needs an Info.plist as a standard. For the `CFBundleDevelopmentRegion` key,  the region will be used in some system API to localize some identifiers like `UIBarButtonSystemItemDone` and `UIBarButtonSystemItemCancel` in pod projects.

Currently, the https://github.com/CocoaPods/CocoaPods/blob/3edf0f3fdc4e34327e5a9507b4d933fd325339b8/lib/cocoapods/generator/info_plist_file.rb#L105 uses the `en` for all the cases, this should be configurable like a property in cocoapods. 

I'm not good at ruby here, sorry! Please confirm this, thanks! Besides, I'm happy for that there are some official `post_install` hooks, too.

**Comment by xingheng** [comment]
#4729

**Comment by dnkoutso** [comment]
Yes that needs to be improved.

**Comment by xingheng** [comment]
I resolved it as fixed with the following post_install hook code temporarily.

```ruby
    post_install do |installer|
        # Update the CFBundleDevelopmentRegion values for resource bundles manually.
        # https://github.com/CocoaPods/CocoaPods/issues/7486
        pod_project_root = File.dirname(installer.pods_project.path)
        info_plist_files = Dir.glob("#{pod_project_root}/**/*Info.plist")

        info_plist_files.each do |info_plist|
            # puts info_plist
            cmd = "/usr/libexec/PlistBuddy '#{info_plist}' -c 'Set CFBundleDevelopmentRegion zh_CN'"

            output = `#{cmd}`
            if $?.exitstatus != 0
                puts output
            end
        end

        puts "\n\e[3m\e[32mUpdated the CFBundleDevelopmentRegion key for #{info_plist_files.length} pod resource bundle(s)!\e[0m\e[23m\n\n"
    end
```

**Comment by dnkoutso** [comment]
will do for now but I will fix it for 1.5.0

**Comment by xingheng** [comment]
Thanks @dnkoutso , I think the right fix is to read the region value from the main app target's `Info.plist` and set it to all the dependent pods' resource bundles, it seems all the Info.plist's name have a format like `ResourceBundle-<PODNAME>-Info.plist` from the static library, the framework bundles have a config file named `Info.plist` within it, too.  Just a reminder. :)

**Comment by dnkoutso** [comment]
@xingheng would be awesome if you can open a PR for it! I think if you learn the project you can make great improvements :)

**Comment by xingheng** [comment]
I dug out a new big hole here when trying to fix it. The `INFOPLIST_FILE` didn't be read in all the components of CocoaPods, that makes sense because the `Info.plist` file could be localized into multiple languages (with multiple subfiles). It's a key to localize, not a result, my thought in above is wrong.

Go back to the localization of `UIBarButtonSystemItemDone`, the prefer localized language is controlled by `preferredLocalizations` of `NSBundle`. [Understanding iOS internationalization](http://maniak-dobrii.com/understanding-ios-internationalization/), this article has a deep explanation.

How to fix it?
The podspec's owners should maintain the supported localization for their resource bundles, `CocoaPods` should support to define it/them in/like `resource_bundles` in podspec. I saw it's defined in https://github.com/CocoaPods/Core/blob/37ed7bbb6b822a1dfee580dc5e1facda418fe31c/lib/cocoapods-core/specification/dsl.rb#L1185 with type `[String, Array]`, this isn't enough to describe it, there should have `[String, Hash]` to expand the bundle properties including paths, localized language list, development region, etc.

From this point, `CocoaPods` didn't support localization too much, there will be plenty of discussion and code needed to implement it. This should be confirmed by maintainers, not a new ruby guy like me. I'll watch it related later. :)

**Comment by dnkoutso** [comment]
@xingheng in the first comment you had mentioned to read the value from the app target and apply that while in your second comment it seems you are arguing to let the podspec author to specify it instead.

Which one is it?

**Comment by xingheng** [comment]
Exactly, it's not just only the `CFBundleDevelopmentRegion` key but the entire `Info.plist` file, which should be localized into multiple languages. The podspec author should specify the localization language list including the corresponding resource bundles, then `cocoapods` parses it and generates the localized `Info.plist` for those bundles.

**Comment by k4ety** [comment]
The original issue [https://github.com/CocoaPods/CocoaPods/pull/4729](url) seems to have been regressed in the latest beta release, I am guessing due to these changes?
